### ISSUE TITLE
Empty host crashes aiohttp

### ISSUE THREAD

**Issue by Insoleet** [aio-libs/aiohttp#510]
Hello,

The following line produce a crash : 

``` python
        response = yield from aiohttp.get("http://:8001/crash")
```

The stack trace is the following : 

```
  File "/usr/lib/python3.4/asyncio/tasks.py", line 238, in _step
    result = next(coro)
  File "/usr/lib/python3.4/site-packages/aiohttp/client.py", line 474, in get
    ret = yield from request(hdrs.METH_GET, url, **kwargs)
  File "/usr/lib/python3.4/site-packages/aiohttp/client.py", line 466, in request
    read_until_eof=read_until_eof)
  File "/usr/lib/python3.4/site-packages/aiohttp/client.py", line 129, in request
    loop=self._loop, response_class=self._response_class)
  File "/usr/lib/python3.4/site-packages/aiohttp/client_reqrep.py", line 77, in __init__
    self.update_host(url)
  File "/usr/lib/python3.4/site-packages/aiohttp/client_reqrep.py", line 118, in update_host
    host = host.encode('idna').decode('utf-8')
AttributeError: 'NoneType' object has no attribute 'encode'
```

**Comment by kxepal** [comment]
What did you expect? (:

**Comment by Insoleet** [comment]
Well shouldn't it raise a more comprehensive error ? 
Like this one : https://github.com/KeepSafe/aiohttp/blob/master/aiohttp/client_reqrep.py#L111

**Comment by kxepal** [comment]
Should be easy to fix. Want to submit PR?

**Comment by Insoleet** [comment]
Ok I'll do. Shoud l write a unit test for this bug ?

**Comment by kxepal** [comment]
Yes. We don't want to have your changes accidentally get broken in future, do we?

**Comment by Insoleet** [comment]
Sure ;) I'll try to work on it tonight.

**Comment by kxepal** [comment]
Thanks! (:

**Comment by asvetlov** [comment]
Fixed by #511
